Makefile

```diff
@@ -452,18 +452,24 @@ build/toolchain/bin/protoc-gen-go$(EXE_EXTENSION):
 	$(GO) install github.com/golang/protobuf/protoc-gen-go
 	mv $(GOPATH)/bin/protoc-gen-go$(EXE_EXTENSION) build/toolchain/bin/protoc-gen-go$(EXE_EXTENSION)
 
+build/toolchain/bin/protoc-gen-grpc-gateway$(EXE_EXTENSION):
+	$(GO) get github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
+	$(GO) install github.com/grpc-ecosystem/grpc-gateway/protoc-gen-grpc-gateway
+	mv $(GOPATH)/bin/protoc-gen-grpc-gateway$(EXE_EXTENSION) build/toolchain/bin/protoc-gen-grpc-gateway$(EXE_EXTENSION)
+
 all-protos: golang-protos mmlogic-simple-protos php-protos
 golang-protos: internal/pb/backend.pb.go internal/pb/frontend.pb.go internal/pb/function.pb.go internal/pb/messages.pb.go internal/pb/mmlogic.pb.go
 internal/pb/%.pb.go: api/protobuf-spec/%.proto build/toolchain/bin/protoc$(EXE_EXTENSION) build/toolchain/bin/protoc-gen-go$(EXE_EXTENSION)
 	$(PROTOC) $< \
 		-I $(CURDIR) -I $(PROTOC_INCLUDES) \
+		-I $(GOPATH)/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis -I $(GOPATH)/src \
 		--go_out=plugins=grpc:$(REPOSITORY_ROOT)
 
-php-protos: examples/functions/php/mmlogic-simple/proto/
 examples/functions/php/mmlogic-simple/proto/:
 	mkdir -p examples/functions/php/mmlogic-simple/proto/
 	$(PROTOC) api/protobuf-spec/messages.proto \
 		-I $(CURDIR) -I $(PROTOC_INCLUDES) \
+		-I $(GOPATH)/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis -I $(GOPATH)/src \
```

Comment:
This regresses the effort to move way from Go workspace setup.

What files are necessary here? Much of them are in $(PROTOC_INCLUDES) directory. Is there something missing there? https://github.com/grpc-ecosystem/grpc-gateway/tree/master/third_party/googleapis

If anything overlay the source repositories data on top of the include directory. You can download the zip of the repository. This would be a separate build target.

Something like this should work.

```Makefile
build/toolchain/bin/protoc$(EXE_EXTENSION):
	... APPEND THE FOLLOWING LINES ...
	mkdir -p $(TOOLCHAIN_DIR/googleapis-temp/
	curl -o $(TOOLCHAIN_DIR)/googleapis-temp/googleapis.zip -L https://github.com/googleapis/googleapis/archive/master.zip
	(cd $(TOOLCHAIN_DIR/googleapis-temp/); unzip -o googleapis.zip)
	cp -rf $(TOOLCHAIN_DIR)/googleapis-temp/googleapis-master/google/rpc/ $(TOOLCHAIN_DIR)/protoc/includes/google/
	rm -rf $(TOOLCHAIN_DIR/googleapis-temp)
```

Reply:
Looking at the rest of the code /rpc/ is not referenced anywhere. I'd only add this Makefile dependency if the /rpc/ directory is necessary through transitive include.

Reply:
Adding REST proxy via gRPC gateway requires 'google/api/annotations.proto' file. You are right, I'll update the commit such that it only changes the dependency part.